firmware/library/L3_HAL/Piezo_Driver.hpp

```diff
@@ -1,11 +1,45 @@
+// Usage:
+// // initialize buzzer with pwm pin number
+// BuzzerInterface* buzzer = new Buzzer(1);
+// buzzer->Beep(500, 0.9); //frequency = 500Hz, Volume = 0.9
 #pragma once
 #include <cstdint>
+#include "../L1_Drivers/pwm.hpp"
 
 class BuzzerInterface
 {
  public:
+    BuzzerInterface(){}
     virtual void Initialize() = 0;
     virtual void Stop() = 0;
-    virtual void Beep(uint32_t frequency, uint32_t volume) = 0;
+
+        // beep continuously @ a certain volume &
+        // @ a specified frequency
+    virtual void Beep(uint32_t frequency, float volume) = 0;
 };
 
+class Buzzer : public BuzzerInterface
+{
+ private:
+    Pwm pwm_;
+
+ public:
+    explicit Buzzer(uint8_t select_pin): pwm_(select_pin){}
+
+    void Initialize() override
+    {
+        // nothing to do here
+    }
+
+    void Stop() override
+    {
+        pwm_.SetDutyCycle(0.0);
+    }
+
+    void Beep(uint32_t frequency = 500, float volume = 1.0) override
+    {
+        pwm_.Initialize(frequency);
```

Comment:
I tried doing that in the very beginning. 
In main.cpp, Id call Initialize and then the Beep function, but it would give me an error because it wouldn't recognize the fact that I had Initialized the PWM in my Initialize function.